name_tracker.go

```diff
@@ -6,14 +6,47 @@ package quamina
 // Quamina's MatchesForEvent API is strongly related to the number of fields it has to look at.  Therefore, a
 // Flattener should call NameTracker for each field it encounters and if it returns false, skip that field
 // and not add it to the flattened output. Here's a clarifying example; consider this JSON event:
-//  {"a": {"b": 1, "c": 2}}
+//
+//	{"a": {"b": 1, "c": 2}}
+//
 // Assuming NameTracker always returns positive, the flattened output should look like
-//  "a\nb" 1
-//  "a\nc" 2
+//
+//	"a\nb" 1
+//	"a\nc" 2
+//
 // Note that the values stored in NameTracker are not the full paths "a\nb" and "a\nc", but the path segments
 // "a", "b", and "c".  This reduces the work the Flattener has to do - whenever it processes an object, it need
 // only look up the member names in the NameTracker. A little thought reveals that this can produce false
 // positives and some potential wasted work, but is a good trade-off.
 type NameTracker interface {
 	IsNameUsed(label []byte) bool
+
+	// This should be actually part of matcher interface and not NameTracker.
```

Comment:
Why not put it in the matcher interface then? We have not done a 1.0 release, we can change APIs if we want to.

Reply:
Will move 👍

Reply:
I remembered now why I did it like this! It means that `Flatten` will need to accept a `Matcher` interface instead.

Maybe a composed interface -
```
type Tracker interface {
  NameTracker
  HierarchalNameTracker
}
```

Or - 
```
type Tracker interface {
  GetNameTracker()
  GetHierarchalNameTracker()
}
```

?